tools/docker/test-image/Dockerfile

```diff
@@ -1,21 +1,24 @@
-FROM ubuntu:latest
+FROM ubuntu:20.04
 
 RUN \
+        export DEBIAN_FRONTEND=noninteractive && \
         dpkg --add-architecture i386 && \
         apt-get update -qq && \
-        apt-get install -y nasm gdb unzip p7zip-full openjdk-8-jre wget python python3 qemu-system-x86 git-core build-essential libc6-dev-i386-cross libc6-dev-i386 && \
-        wget https://nodejs.org/dist/v8.9.4/node-v8.9.4-linux-x64.tar.xz && \
-        tar xfv node-v8.9.4-linux-x64.tar.xz && \
-        rm node-v8.9.4-linux-x64.tar.xz && \
+        apt-get dist-upgrade -y
+RUN \
+        export DEBIAN_FRONTEND=noninteractive && \
+        apt-get install -y nasm gdb unzip p7zip-full openjdk-8-jre wget python python3 qemu-system-x86 git-core build-essential libc6-dev-i386-cross libc6-dev-i386 clang curl
+RUN \
+        NODEVSN=v14.16.0 && \
+        wget https://nodejs.org/dist/$NODEVSN/node-$NODEVSN-linux-x64.tar.xz && \
+        tar xfv node-$NODEVSN-linux-x64.tar.xz && \
+        rm node-$NODEVSN-linux-x64.tar.xz && \
+        echo 'export PATH="$PATH:/node-$NODEVSN-linux-x64/bin"' >> ~/.bashrc && \
```

Comment:
I think it might be better to use the version of Node that comes with Ubuntu 20.04 these days. Could you try that? I'll fix any scripts that are broken (if any) afterwards.

Reply:
Willdo